src/libcore/fmt/mod.rs

```diff
@@ -180,8 +180,9 @@ enum Void {}
            issue = "0")]
 #[doc(hidden)]
 pub struct ArgumentV1<'a> {
-    value: &'a Void,
-    formatter: fn(&Void, &mut Formatter) -> Result,
+    _ph: PhantomData<&'a ()>,
```

Comment:
It might have had something with old trans' `_match` codegen using `is_uninhabited`, directly or indirectly.
TBH, any old call to `is_uninhabited` is suspicious. You also may have had more local changes at the time.

If we change any behavior here, we *should* warn about non-generic types that can't have values, IMO.
Otherwise we risk a lot of subtle breakage across the ecosystem (but this is more relevant to the `Error` PR).